packages/next/src/client/app-dir/link.tsx

```diff
@@ -193,9 +193,11 @@ function mountLinkInstance(
     // case there's a logical error somewhere else.
     unmountLinkInstance(element)
   }
-  links.set(element, instance)
-  if (observer !== null) {
-    observer.observe(element)
+  if (element) {
```

Comment:
We should probably move this check to the start of the function since none of it is relevant if the element is falsey